### PR TITLE
Refactor BSP tests to work in lxc container.

### PR DESCRIPTION
**Pull request by petedyerarm**
Changed the BSP tests to run under pytest control. This removes the need
for the linaro repo.

### CODE REVIEW

**Comment by petedyerarm** [comment]
synchronize with https://github.com/ARMmbed/mbl-tools/pull/330

File: ci/lava/tests/bsp.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    parameters:
+        #
+        # virtual_env: specifies the Python virtual environment
+        #
+        virtual_env:
+
+run:
+    steps:
+        - . $virtual_env/bin/activate
+
```
Comment (diegorusso) [inline ci/lava/tests/bsp.yaml]:
```suggestion
```

File: ci/lava/tests/test-bsp.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Script to run systemd tests on the DUT."""
+
+import pytest
+import re
+import subprocess
+
+
+class TestBsp:
```
Comment (diegorusso) [inline ci/lava/tests/test-bsp.py]:
```suggestion
class TestBSP:
```

File: ci/lava/tests/test-bsp.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    """Class to encapsulate the testing of systemd on a DUT."""
+
+    dut_address = ""
+
+    def test_setup_dut_addr(self, dut_addr):
+        """Store the device address."""
+        TestBsp.dut_address = dut_addr
+
+        assert dut_addr != ""
```
Comment (diegorusso) [inline ci/lava/tests/test-bsp.py]:
```suggestion
        assert dut_addr
```

File: ci/lava/tests/test-bsp.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    def test_badblocks(self, execute_helper):
+        """Perform the test on the DUT via the mbl-cli."""
+        err, stdout, stderr = execute_helper.send_mbl_cli_command(
+            [
+                "shell",
+                'sh -l -c "badblocks -v '
+                "$(mount | grep \"on \/ type\" | cut -d' ' -f 1)"
```
Comment (diegorusso) [inline ci/lava/tests/test-bsp.py]: This line fails the sanity check. I think it's enough to do so (please test it):
```suggestion
                r"$(mount | grep \"on \/ type\" | cut -d' ' -f 1)"
```

Comment (petedyerarm) [reply ci/lava/tests/test-bsp.py]: changed to use /sbin/blkid instead.

**Pull request merged**